Assignment1/RegExpr.cs

```diff
@@ -1,10 +1,65 @@
 namespace Assignment1;
 
+using System.Text.RegularExpressions;
+
 public static class RegExpr
 {
-    public static IEnumerable<string> SplitLine(IEnumerable<string> lines) => throw new NotImplementedException();
+    public static IEnumerable<string> SplitLine(IEnumerable<string> lines)
+    {
+        var splitPattern = @"[^\w\d]";
+        var matchPattern = @"[\w*\d*]";
+
+        foreach (var s in lines)
+        {
+            foreach (var w in Regex.Split(s, splitPattern))
+            {
+                if (Regex.Match(w, matchPattern).Success)
+                {
+                    yield return w;
+                }
+            }
+        }
+    }
+
+    public static IEnumerable<(int width, int height)> Resolution(string resolutions)
+    {
+        var splitPattern = @"[^\w\d]";
+        var matchPattern = @"((?<width>\d+)x(?<height>\d+))";
+
+        var splittedResolutions = Regex.Split(resolutions, splitPattern);
+
+        foreach (var res in splittedResolutions)
+        {
+            foreach (Match match in Regex.Matches(res, matchPattern))
+            {
+                yield return (int.Parse(match.Groups["width"].Value), int.Parse(match.Groups["height"].Value));
+            }
+        }
+    }
+
+    public static IEnumerable<string> InnerText(string html, string tag)
+    {
+        var pattern = $@"<(?<tag>{tag}).*?>(?<text>.*?)<\/\1>";
```

Comment:
Det kan generelt være ret farligt at bruge ´.´ i regex, siden man gør ens constraints så bløde, at den nemt kan tage imod for meget. Men det burde gerne virke så vidt jeg kan se. Evt. erstat med ´[\w\d]*´